wagtail_localize_rws_languagecloud/management/commands/update_translated_pages.py

```diff
@@ -0,0 +1,128 @@
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            dest="dry_run",
+            default=False,
+            help="Don't actually sync or send the pages to RWS LanguageCloud",
+        )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        # Set up logging
+        log_level = logging.INFO
+        if options["verbosity"] > 1:
+            log_level = logging.DEBUG
+
+        logger = logging.getLogger(__name__)
+
+        console = logging.StreamHandler()
+        console.setLevel(log_level)
+        console.setFormatter(logging.Formatter("%(levelname)s - %(message)s"))
+        logger.addHandler(console)
+        logger.setLevel(log_level)
+
+        logger.info("Updating translations of pages with stale translated content...")
+
+        # Get all pages that have been published _after_ the translation source was synced.
+        pages = Page.objects.filter(
+            translation_key=OuterRef("object_id"), locale_id=OuterRef("locale_id")
+        )
+        all_sources = TranslationSource.objects.annotate(
+            last_published_at=Subquery(pages.values("last_published_at")[:1]),
+            page_pk=Subquery(pages.values("id")[:1]),
+            page_title=Subquery(pages.values("title")[:1]),
+        ).filter(
+            # Only sources for live pages
+            object_id__in=Page.objects.live().values_list("translation_key", flat=True),
+            # Only pages that have been published after the last sync
+            last_updated_at__lt=F("last_published_at"),
+        )
+
+        sources_to_update = all_sources.exclude(
+            # Only pages without ongoing translation projects,
+            languagecloudproject__lc_project_status__in=[
+                LanguageCloudStatus.CREATED,
+                LanguageCloudStatus.IN_PROGRESS,
+            ]
+        )
+
+        total_count = len(all_sources)
+        to_update_count = len(sources_to_update)
+
+        logger.info(
+            f"Found {total_count} page(s) with stale translated content.\n"
+            f"       {to_update_count} will be synced.\n"
+            f"       {total_count - to_update_count} will be skipped."
+        )
+
+        all_pages = {(source.page_pk, source.page_title) for source in all_sources}
+
+        updated_pages = set()
+
+        for index, source in enumerate(sources_to_update):
+            updated_pages.add((source.page_pk, source.page_title))
+
+            counter = f"[{index + 1}/{to_update_count}]"
+
+            if options["dry_run"]:
+                logger.info(
+                    f'{counter} Would update translations for page: {source.page_pk} - "{source.page_title}"'
+                )
+                continue
+
+            logger.info(
+                f'{counter} Updating translations for page: {source.page_pk} - "{source.page_title}"'
+            )
+
+            # Sync content to translated pages
+            source.update_from_db()
```

Comment:
I might be missing something or not understanding, but something looks off here. This doesn't look like it will sync content to translated pages - it will just re-fetch the object from the DB - right?

i.e: this command will create a `LanguageCloud` project, but it won't actually sync the target locale on the page object like the docs say? Maybe we also need a test covering this.

Reply:
This confused the heck out of me too.

You're thinking of `refresh_from_db()` which is a Django method. This is `update_from_db()` which comes from Wagtail Localize. [Ref](https://github.com/wagtail/wagtail-localize/blob/f1301493b6dd2dcd7df3c75f46c6fb7170f0186e/wagtail_localize/models.py#L448)

Reply:
Yes that is exactly what I was (wrongly) thinking :smile: